src/main/java/net/snowflake/ingest/streaming/internal/ChannelCache.java

```diff
@@ -33,6 +40,12 @@ void addChannel(SnowflakeStreamingIngestChannelInternal<T> channel) {
         this.cache.computeIfAbsent(
             channel.getFullyQualifiedTableName(), v -> new ConcurrentHashMap<>());
 
+    // Update the last flush time for the table, add jitter to avoid all channels flush at the same
+    // time when the blobs are not interleaved
+    this.lastFlushTime.putIfAbsent(
```

Comment:
not sure i understand what this helps with. If someone does addChannel, and doesn't add any data for a minute, the first row that they add will trigger a flush since we'll mistakenly think its been a long time since the last flush.

Reply:
Yes. Should we change the logic to following?
1. Set `lastFlushTime` to null when create a channel.
2. When calling `putRow` or `putRows`, if `lastFlushTime` is null, set to current time.
3. Whenever a table is flushed, set `lastFlushTime` to null, go to step 2.

Reply:
as discussed pl track this with a JIRA so we don't forget about it.

Reply:
[Jira](https://snowflakecomputing.atlassian.net/browse/SNOW-1531922) created.